src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -155,6 +159,15 @@ module.exports = function<T, P, I, C>(config : HostConfig<T, P, I, C>) {
       } else if (returnFiber) {
         // If there's no more work in this returnFiber. Complete the returnFiber.
         workInProgress = returnFiber;
+        // If we're stepping up through the child, that means we can now commit
+        // this work. We should only do this when we're stepping upwards because
+        // completing a downprioritized item is not the same as completing its
+        // children.
+        if (workInProgress.childInProgress) {
```

Comment:
This let me get rid of the wasDeprioritized flag. We only do work when stepping backup instead of when we're completing a node that was deprioritized (with a "hidden" attribute).